docs/api/matchers.md

```diff
@@ -117,3 +135,21 @@ On iOS, matched elements are sorted by their x and y axes.
 ```js
 element(by.text('Product')).atIndex(2);
 ```
+
+## Regex matching
+
+For supported matchers ([`id`](#byidid), [`label`](#bylabellabel), [`text`](#bytexttext)), you can also utilize regex (Regular Expressions) alongside certain flags. Here's a table with the supported flags:
+
+| Flag | Name          | Modification                                                                                                                                             |
+|------|---------------|----------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `i`  | Ignore Casing | Makes the expression search case-insensitively.                                                                                                          |
+| `s`  | Dot All       | Makes the wild character `.` match newlines as well.                                                                                                     |
+| `m`  | Multiline     | Makes the boundary characters (`^` and `$`) match the beginning and ending of every single line instead of the beginning and ending of the whole string. |
+
+**Note:** JS flags like `g` (global) and `y` (sticky) that are not supported, as well as `u` (unicode) which is always implied, are ignored when parsing input.
```

Comment:
I'd suggest to create an admonition and warn about possible issues with look-behind expressions (iOS/Swift doesn't support them where as Kotlin supports), e.g.:

![image](https://github.com/wix/Detox/assets/1962469/11c7ea36-f5ad-4cf3-b313-0edd55d63577)

Reply:
@noomorph sounds good. Great catch with the "look behind" assertions! IMO we should provide this information inline.

```suggestion
:::caution Note

Regular expression flags such as `g` (global) and `y` (sticky) that are not supported, as well as `u` (unicode) which is always implied, are ignored when parsing input. 

Pay attention that as of writing this note, Android supports lookbehind assertions in its regular expression implementation, while iOS does not. It's advisable to check the platform-specific documentation for limitations.

:::
```

Reply:
Due to the fact that this is only a button tap (+ to avoid the bikeshedding), I will allow myself to commit this change 🙂